[CDO] Patches to support Phoenix

stephenneuendorffer opened the pull request.

stephenneuendorffer committed:
[CDO] Patches to support Phoenix

jackl-xilinx committed:
Fixed ecc for IPU support

stephenneuendorffer committed:
Merge pull request #1 from jackl-xilinx/phoenix_v2023.2

Fixed ecc for IPU support

keryell reviewed (commented):
A comment on why/what this does since it is open-source or this is just a firmware secret detail?

makslevental committed:
add decls

stephenneuendorffer committed:
Merge pull request #3 from makslevental/dead_code

Add forward decls for `cdo_*`

eddierichter-amd committed:
Add VCK5000 support

Building the projects creates .o files, .so files and copies some headers. None
of these artifacts are tracked under source control so we don't need to see
them.

* [amdair] xaie: Enable backend selection at init time

All IO backends are compiled no matter the configuration. The only
thing that changes is the XAIE_DEFAULT_BACKEND macro that points to
the correct backend as specified by a compile-time flag. There is no
reason why the backend selection cannot be made at runtime, giving
more control to the application as to how it wants to use the
library.

Add a configuration item specifying the backend. When the library is
initialized, the backend will be chosen based on this parameter. If
the user-selected backend index is out of range, the library will
fall back to the existing default backend logic.

* [amdair] driver: add amdair backend

The I/O commands are sent to the device using the amdair driver's
sysfs interface. This requires the amdair driver to be loaded before
sending any commands.

---------

Signed-off-by: Joel Nider <joel.nider@amd.com>
Co-authored-by: Joel Nider <joel.nider@amd.com>

stephenneuendorffer committed:
Fix initialization warning (#6)

The XAie_InstDeclare define declares a structure, but didn't completely
initialize it.  This patch initializes the structure to a default
value.

jnider committed:
driver: Fix warnings (#5)

There are several trivial warnings that appear when building. Address
them as necessary by fixing the problems. In general, unused parameters
to functions that are part of an interface cannot be removed so they are
marked with the UNUSED macro. Signed comparisons can never be negative
and uints can never be larger than UINT_MAX. XAIE columns are generally
a u8 type (limited to 256 columns) although stored in a u32 in places.
If we ever need to support more than 256 columns this needs to change
across the board.

Signed-off-by: Joel Nider <joel.nider@amd.com>

gabrielrodcanal committed:
Fix name of external symbol AmdairBackend. Otherwise the shared library will have an undefined symbol and the mlir-air tests will fail to compile (#8)

erieaton-amd committed:
Fix build errors on MSVC

* Removed illegal cast to struct type
* Replaced variable length array with malloc
* Added required `=` to array initialization with designator
* Removed use of unistd.h
* Replaced usage of pthread.h
* Moved the alignment macro to avoid a circular dependency
* Disabled use of gcc-style attribute
* Replaced illegal arithmetic on void pointer

makslevental committed:
add <unistd.h> to xaie_amdair.c

makslevental committed:
free RscArrPerTile

makslevental committed:
tabs not spaces

makslevental committed:
Merge pull request #9 from erieaton-amd/msvc-fixes

Fix build errors on MSVC

erieaton-amd committed:
Load ELF file as binary on windows

makslevental committed:
Merge pull request #11 from erieaton-amd/w32-fixes

Load ELF file as binary on windows
